web/packages/teleterm/src/services/tshd/cluster.ts

```diff
@@ -65,3 +72,31 @@ export function proxyHostname(proxyHost: string) {
 
   return whatwgURL.hostname;
 }
+
+/** Produces cluster with properties that can be read from the profile. */
+export function makeClusterWithOnlyProfileProperties(a: Cluster): Cluster {
```

Comment:
would it be beneficial to have this as a separate type (something produced with some typescript utility type?). i dont mind either way. also, we setting fields as `undefined` on purpose to be extra verbose and readable?